cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -139,18 +139,17 @@ void reader::impl::decode_page_data(size_t skip_rows, size_t num_rows)
   chunk_nested_valids.host_to_device(_stream);
   chunk_nested_data.host_to_device(_stream);
 
-  if (total_str_dict_indexes > 0) {
-    gpu::BuildStringDictionaryIndex(chunks.device_ptr(), chunks.size(), _stream);
-  }
-
   gpu::DecodePageData(pages, chunks, num_rows, skip_rows, _stream);
+
+  _stream.synchronize();
+
   pages.device_to_host(_stream);
   page_nesting.device_to_host(_stream);
   _stream.synchronize();
```

Comment:
The first synchronize seems unnecessary. 
```suggestion
  pages.device_to_host(_stream);
  page_nesting.device_to_host(_stream, true);
```